### DISCUSSION TITLE
How about adding swift-call that creates a frame?

### DISCUSSION THREAD

**Discussion by BECATRUE** [snu-quiqcl/qiwis#148] [Ideas]
In current system, we can craate a new app using `self.swiftcall.createApp(...)`.
However, we cannot create a new frame in an existing app.

For example, there is an app `APP1` and its frames are `FRAME1` and `FRAME2`, but `APP1.frames()` returns only `FRAME1`.
In this case, an user cannot open `FRAME2` at any time.

So, how about adding swift-call `createFrame`?

**Comment by BECATRUE** [comment]
Please leave your opinion freely, @kangz12345.

**Comment by kangz12345** [comment]
I agree with the necessity, but the frames cannot be passed via string messages.
How about adding a swift-call for updating frames, e.g., `updateFrames(name: str)`?
When this is called, `Swift` calls `.frames()` of the app `name`, and then adds the new frames and deletes the removed frames.
Then the app will do the following to create or destroy some frames:
1. Update the iterable which is returned by `self.frames()`.
2. Call the swift-call `updateFrames(name)` with its name.

**Reply by BECATRUE** [reply]
Oh, I have a similar idea to yours!

To add a little bit more functionality,
- In default, add only the new frames (remain the original frames)
- With the argument `update` is `True`, add the new frames and also update the original frames

```python
self.swiftcall.updateFrames(name)               # 1st case
self.swiftcall.updateFrames(name, update=True)  # 2nd case
```

What do you think of this method?

**Reply by kangz12345** [reply]
What do you mean by to update the original frames?

**Reply by kangz12345** [reply]
Oh, you explained it in #150.
